### PR TITLE
improve operator install

### PR DESCRIPTION
**Pull request by maleck13**
This can be reviewed but I need to fix the tests.

Verification 

Start the operator locally
set the flag for products to just use solution-explorer  ``` operator-sdk up local --namespace=i1 --operator-flags="--products=solution-explorer" ```
create installation

The solution explorer install should complete

delete the subscription
the solution explorer should be re installed

delete the ns
the solution explorer should be re installed

### CODE REVIEW

File: pkg/controller/installation/marketplace/manager.go [hunk 1]
```diff
@@ -41,7 +43,7 @@ func GetOperatorSources() *operatorSources {
 
 //go:generate moq -out MarketplaceManager_moq.go . MarketplaceInterface
 type MarketplaceInterface interface {
-	CreateSubscription(ctx context.Context, serverClient pkgclient.Client, owner ownerutil.Owner, os marketplacev1.OperatorSource, ns string, pkg string, channel string, operatorGroupNamespaces []string, approvalStrategy coreosv1alpha1.Approval) error
+	InstallOperator(ctx context.Context, serverClient pkgclient.Client, owner ownerutil.Owner, os marketplacev1.OperatorSource, t Target, operatorGroupNamespaces []string, approvalStrategy coreosv1alpha1.Approval) error
 	GetSubscriptionInstallPlan(ctx context.Context, serverClient pkgclient.Client, subName, ns string) (*coreosv1alpha1.InstallPlan, *coreosv1alpha1.Subscription, error)
```
Comment (sedroche) [inline pkg/controller/installation/marketplace/manager.go]: This works, as in we need to wait for the InstallPlan to be sure the CRDs are in the cluster. What we really want to do to sequence things correctly is to wait for when the CSV is ready.

The subscription has a `status.currentCSV` prop and a CSV has a `status.phase` = Succeeded which should be able to be used to to this

File: pkg/controller/installation/marketplace/manager.go [hunk 2]
```diff
@@ -51,101 +53,112 @@ func NewManager() *MarketplaceManager {
...
 	}
-	ownerutil.EnsureOwner(csc, owner)
-
-	// check catalog source config status
-	cscErr := serverClient.Get(ctx, pkgclient.ObjectKey{Namespace: csc.Namespace, Name: csc.Name}, csc)
-	if cscErr != nil && !k8serr.IsNotFound(cscErr) {
-		return errors.Wrap(cscErr, "failed to get catalog source config "+csc.Name)
+	scopy := sub.DeepCopy()
+	if err := client.Get(ctx, pkgclient.ObjectKey{Namespace: sub.Namespace, Name: sub.Name}, scopy); k8serr.IsNotFound(err) {
+		if err := client.Delete(ctx, csc, func(options *pkgclient.DeleteOptions) {
```
[the remark this answers is not in the document]
Comment (sedroche) [reply pkg/controller/installation/marketplace/manager.go]: I also checked (using the UI not the operator) that the subscription can be created first, and the operator installation still succeeds. So in that case the subscription can be set as owner on the catalogsourceconfig and operatorgroup.
In that case if:
```go
if err := client.Get(ctx, pkgclient.ObjectKey{Namespace: sub.Namespace, Name: sub.Name}, scopy); k8serr.IsNotFound(err) {
```

we can be sure the csc and og are not.

[the remark this answers is not in the document]
Comment (maleck13) [reply pkg/controller/installation/marketplace/manager.go]: this seems a good approach it also removes the need for delete perms on the catalogsourceconfig

[the remark this answers is not in the document]
Comment (maleck13) [reply pkg/controller/installation/marketplace/manager.go]: @sedroche I have updated with the new approach

File: pkg/controller/installation/marketplace/manager.go [hunk 3]
```diff
@@ -51,73 +54,81 @@ func NewManager() *MarketplaceManager {
...
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: ns,
-			Name:      pkg,
-		},
-	}
+type Target struct {
+	Namespace, Pkg, Channel string
+}
 
+func (m *MarketplaceManager) createAndWaitCatalogSourceConfig(ctx context.Context, owner ownerutil.Owner, t Target, os marketplacev1.OperatorSource, client pkgclient.Client, sub *coreosv1alpha1.Subscription) (string,error) {
```
Comment (sedroche) [inline pkg/controller/installation/marketplace/manager.go]: This should be `createAndWaitCatalogSource` now. And I don't think `sub` is used.

**Pull request merged**